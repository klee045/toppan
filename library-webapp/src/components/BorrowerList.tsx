import React, { CSSProperties } from "react";
import Borrower from "./Borrower";

const BorrowerListStyle: CSSProperties = {
  width: "inherit",
};

const BorrowerList = ({
  borrowers,
}: {
  borrowers: string[];
}): React.ReactElement => {
  return (
    <div id="borrowerList" style={BorrowerListStyle}>
      {borrowers.map((borrower, index) => (
        <div key={index + 1}>
          <Borrower name={borrower} />
        </div>
      ))}
    </div>
  );
};

export default BorrowerList;
