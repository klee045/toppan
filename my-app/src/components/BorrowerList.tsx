import React, { CSSProperties } from "react";
import Borrower from "./Borrower";

const BorrowerListStyle: CSSProperties = {
  width: "inherit",
};

const BorrowerList = ({
  borrowers,
}: {
  borrowers: string[];
}): React.ReactElement => {
  return (
    <div id="borrowerList" style={BorrowerListStyle}>
      {borrowers.map((borrower) => (
        <Borrower name={borrower} />
      ))}
    </div>
  );
};

export default BorrowerList;
