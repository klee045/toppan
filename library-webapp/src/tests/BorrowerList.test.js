import React from "react";
import renderer from "react-test-renderer";
import BorrowerList from "../components/BorrowerList";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = renderer.create(<BorrowerList borrowers={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('has a div with id "borrowerList"', () => {
  const tree = renderer.create(<BorrowerList borrowers={[]} />);
  expect(
    shallow(<BorrowerList borrowers={[]} />)
      .find("div#borrowerList")
      .exists()
  ).toBe(true);
});
