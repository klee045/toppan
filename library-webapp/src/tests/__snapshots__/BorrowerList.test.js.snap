// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="borrowerList"
  style={
    Object {
      "width": "inherit",
    }
  }
/>
`;
